src/FakeItEasy/Core/CompletedFakeObjectCall.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace FakeItEasy.Core
+{
+    using System.Linq;
+    using System.Reflection;
+    using FakeItEasy.Configuration;
+
+    internal class CompletedFakeObjectCall : ICompletedFakeObjectCall
+    {
+        public CompletedFakeObjectCall(object fakedObject, MethodInfo method, object[] arguments, object returnValue)
+        {
+            this.FakedObject = fakedObject;
+            this.Method = method;
+            this.Arguments = new ArgumentCollection(arguments, method);
+            this.ReturnValue = returnValue;
+        }
+
+        public object ReturnValue { get; }
+
+        public MethodInfo Method { get; }
+
+        public ArgumentCollection Arguments { get; }

```

Comment:
It's a little troubling that `Arguments` means two different things, depending on the type of the underlying object implementing `IFakeObjectCall`. I'm not sure what to do about it, though.